File: python/src/Conjured.py
from src.Item import Item
from src.updateable import updateable

class Conjured(Item, updateable):
    def __init__(self, name, sell_in, quality):
        Item.__init__(self, name, sell_in, quality)
    
    def getSell_in(self):
        return self.sell_in

    def getQuality(self):
        return self.quality
    
    def updateQuality(self):
        if self.sell_in <= 0:
            self.quality -= 4
        else:
            self.quality -=2
        self.checkQuality()
        self.sell_in -= 1

    def checkQuality(self):
        if self.quality <= 0:
            self.quality = 0
        elif self.quality >= 50:
            self.quality = 50

File: java/src/main/java/edu/poniperro/gildedrose/Conjured.java
package edu.poniperro.gildedrose;

public class Conjured extends Item implements Updateable {

    public Conjured(String name, int sell_in, int quality) {
        super(name, sell_in, quality);
    }

    @Override
    public void updateQuality() {
        if (getSell_in() <= 0) {
            setQuality(getQuality() - 4);
        } else {
            setQuality(getQuality() - 2);
        }
        setSell_in();
        checkQuality();
    }

    private void checkQuality() {
        if (getQuality() <= 0) {
            setQuality(0);
        }
    }
}
